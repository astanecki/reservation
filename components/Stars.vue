<template>
  <div class="stars" v-html="stars" />
</template>
<script>
export default {
  name: 'Stars',

  props: {
    number: {
      type: Number,
      required: true,
    },
  },

  data() {
    return {
      filledStar: `&#9733;`,
      emptyStar: `&#x2606;`,
      allStarsNumber: 5,
    };
  },

  computed: {
    stars() {
      let htmlContent = '';
      let i = 0;

      for(i; i < this.allStarsNumber; i++) {
        htmlContent += i <= (this.number - 1) ? `<span class="stars--filled">${this.filledStar}</span>` : this.emptyStar;
      }

      return htmlContent;
    }
  },
}
</script>
<style lang="scss">
.stars {
  &--filled {
    color: #12a67e;
    margin-right: 1px;
  }
}
</style>
