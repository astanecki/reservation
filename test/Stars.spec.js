import { mount } from '@vue/test-utils'
import Stars from '@/components/Stars.vue'

describe('Stars', () => {
  const createWrapper = value => {
    return mount(Stars, {
      propsData: {
        number: value,
      }
    });
  };
  let wrapper;

  it('is a Vue instance', () => {
    wrapper = createWrapper(2);

    expect(wrapper.vm).toBeTruthy();
  });

  it(`should be mounted properly when 0 given`, () => {
    expect(createWrapper(0).vm.$el).toMatchSnapshot();
  });

  it(`should be mounted properly when 1 given`, () => {
    expect(createWrapper(1).vm.$el).toMatchSnapshot();
  });

  // todo implement half star
  xit(`should be mounted properly when 4.5 given`, () => {
    expect(createWrapper(4.5).vm.$el).toMatchSnapshot();
  });

  it(`should be mounted properly when 5 given`, () => {
    expect(createWrapper(5).vm.$el).toMatchSnapshot();
  });
})
