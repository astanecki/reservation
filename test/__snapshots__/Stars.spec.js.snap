// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars should be mounted properly when 0 given 1`] = `
<div
  class="stars"
>
  ☆☆☆☆☆
</div>
`;

exports[`Stars should be mounted properly when 1 given 1`] = `
<div
  class="stars"
>
  <span
    class="stars--filled"
  >
    ★
  </span>
  ☆☆☆☆
</div>
`;

exports[`Stars should be mounted properly when 5 given 1`] = `
<div
  class="stars"
>
  <span
    class="stars--filled"
  >
    ★
  </span>
  <span
    class="stars--filled"
  >
    ★
  </span>
  <span
    class="stars--filled"
  >
    ★
  </span>
  <span
    class="stars--filled"
  >
    ★
  </span>
  <span
    class="stars--filled"
  >
    ★
  </span>
</div>
`;
